wgpu-hal/src/dx12/device.rs

```diff
@@ -221,64 +222,147 @@ impl super::Device {
             .map(|name| ffi::CString::new(name.as_str()).unwrap())
             .map_err(|e| crate::PipelineError::Linkage(stage_bit, format!("{}", e)))?;
 
-        let mut shader_data = native::Blob::null();
-        let mut error = native::Blob::null();
-        let mut compile_flags = d3dcompiler::D3DCOMPILE_ENABLE_STRICTNESS;
+        let empty_str = "";
```

Comment:
Is it necessary to make `empty_str` its own variable here? In fact, you could probably make this entire block:

```rust
let source_name = stage.module.raw_name.as_ref().and_then(|cstr| cstr.to_str()).unwrap_or("");
```